share/completions/pip.fish

```diff
@@ -1 +1,2 @@
-pip completion --fish 2>/dev/null | source
+__fish_cache_sourced_completions pip completion --fish 2>/dev/null
+or pip completion --fish 2>/dev/null | source
```

Comment:
Looks good to me.
It's probably doable to fix `pip` slowness.
They use `pyparsing`; when I used that it needed 2 seconds to parse 1000 lines.

Reply:
Honestly, a *lot* of python commands in particular have startup slowness.

That's also why we try not to launch hg until we absolutely have to.

Not that it matters, we're dealing with these now, as they are.